src/main/scala/io/circe/yaml/Parser.scala

```diff
@@ -1,22 +1,26 @@
 package io.circe.yaml
 
-import Parser._
 import cats.syntax.either._
 import io.circe._
+import io.circe.yaml.Parser._
 import java.io.{ Reader, StringReader }
-import org.yaml.snakeyaml.LoaderOptions
-import org.yaml.snakeyaml.Yaml
+import org.yaml.snakeyaml.{ LoaderOptions, Yaml }
 import org.yaml.snakeyaml.constructor.SafeConstructor
 import org.yaml.snakeyaml.nodes._
 import scala.collection.JavaConverters._
 
+
 final case class Parser(
-  maxAliasesForCollections: Int = 50
+  maxAliasesForCollections: Int = 50,
+  nestingDepthLimit: Int = 50,
+  codePointLimit: Int = 3 * 1024 * 1024 // 3 MB
```

Comment:
Should we add all the other remaining `LoaderOptions` as well while we are at it?

Also, this is certainly going to break bin compat with 0.14.2.  Can you pull latest from master and run `mimaReportBinaryIssues`?

Reply:
I've addressed all bin compat issues, seems OK to me now.

With respect to the other (boolean) `LoaderOptions`, a lengthy investigation we had on this revealed that they are ignored when using the underlying `Yaml.compose` method in snake-yaml - as circe-yaml does. They only affect the behaviour of snake-yaml's `Yaml.loadAs`, which is currently not used by circe-yaml (imo for the best).